internal/daemon/daemon.go

```diff
@@ -800,6 +803,9 @@ func New(opts *Options) (*Daemon, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err = ovld.PassServiceArgs(opts.ServiceArgs); err != nil {
+		return nil, err
+	}
```

Comment:
Instead of adding a `PassServiceArgs` method that's called after `overlord.New()`, why don't we just *pass the service args* directly to `overlord.New()` as a new parameter?

And the same for `ServiceManager.SetServiceArgs()` -- just pass them down via `servstate.NewManager()`.

Reply:
Changed it, thanks. Updated the tests as well.